Pr94x l1t Muon Unpacker Zero Suppression and DQM. Clang cleaning.

rekovic opened the pull request:
PR 94x - Upgrade the Muon Unpacker and DQM to inteligentely handle Zero Suppression, making it BX aware.   Also includes CLANG cleaning 

Needed for data taking and available in the release at P5 before hardware change goes online (soon).
- This is compatible with past, current, and future  data.

vsola committed:
Initial import of the UFSD geometry

forthommel committed:
Updated the geometry testing configuration

thomreis committed:
Add missing BlockHeader flags in Block constructor.

nminafra committed:
Updated Geometry for UFSD

thomreis committed:
Add BxBlock class to hold raw data from one BX.

thomreis committed:
Add a function to get all BX blocks from one Block payload.

thomreis committed:
Upgrade MuonUnpacker to handle MP7 zero suppression by BX mode.

thomreis committed:
Upgrade RegionalMuonGMTUnpacker to handle MP7 zero suppression by BX mode.

thomreis committed:
Upgrade IntermediateMuonUnpacker to handle MP7 zero suppression by BX mode.

thomreis committed:
Make muon collection pointer a member variable.

thomreis committed:
Skip unpacking if the block size is 0 (All BX blocks were zero suppressed).

thomreis committed:
Adapt to changed BX header format.

thomreis committed:
Support for per-BX ZS format.

thomreis committed:
Follow some clang-tidy suggestions.

thomreis committed:
Implement some clang-tidy suggestions

thomreis committed:
Set accidentally changed global tags back to original value.

thomreis committed:
Remove unneeded use of c_str and streams.

uhh2-bot committed:
FIX in GBL reference trajectories: add missing initialisations

Adds missing matrix initialisations to the EIGEN matrices.
This was not needed in previous implementation of GBL based on ROOT TMatrix
(ROOT presets matrices to 0., EIGEN not).

Fix provided by @ckleinw

thomreis committed:
Use modules for threaded processing.

thomreis committed:
add override and remove redundant virtual keywords

thomreis committed:
Remove unneeded include.

makortel committed:
Add SeedStopInfo, include SeedStopReason in it, and produce it instead of SeedStopReason

makortel committed:
Add candidates/seed to SeedStopInfo and propagate to TrackingNtuple

makortel committed:
Minor cleanup

The intention is clearly to issue LogWarning in case a seed collection
is missing, but the current code throws an exception as the Handle is
dereferenced before isValid() check. Therefore I moved the Handle
dereference inside the if(isValid) block.

There is no need to construct RefToBase<TrajectorySeed> as
edm::View<T> has operator[] returning direclty const reference to the
contained object.

In TrackBuildingAnalyzer move the pointer initialization to the class
definition, and initialize bool members in the constructor.

makortel committed:
Monitor seed stopping source and number of candidates per seed in DQM

makortel committed:
Move GBRForest reading as part of the MVA template argument class

Commit:
Correct for geometry access for Hcal

Commit:
Try to satisfy codechecker

jalimena committed:
match regular expression in particle name

lgray committed:
move to timing_layer era

jalimena committed:
apply cmsbuild code checks patch

Commit:
Add a new method to compute energy sum of ECAL towers preceding a HCAL cell

rappoccio committed:
Adding some printout information

makortel committed:
Code style checks

makortel committed:
Seeding region rebuild may return empty set of trajectories

makortel committed:
Code style checks

makortel committed:
Use static_assert for TrajectoryStopReason as well

wddgit committed:
Move functions filling common provenance to IOPool/Provenance

rappoccio committed:
Key comparison now based on final packed candidates... much easier.

dimattia committed:
use of C array removed

fabozzi committed:
adding DQM-only wf on AOD input for HLT validation

dimattia committed:
array accessed with parenthesis operator

mackoo13 committed:
timing detectors geometry (from Nicola Minafra and CTPPS repository)

fabozzi committed:
adding proper HARVESTING for this new wf

prebello committed:
adding SMS-T1tttt_mGl-1500_mLSP-100_13 PU pmx

prebello committed:
adjusting job splittings

prebello committed:
updating GENSIM strings

npostiau committed:
Better indentation

kurtejung committed:
correcting assert for equality

Martin-Grunewald committed:
HLT menu Luna (V3.1) + frozen update

kpedro88 committed:
reject seed candidates not part of superclusters

kpedro88 committed:
code checks

kurtejung committed:
applying code checks

kurtejung committed:
code checks for header applied

Commit:
Simplify eECALMatrix a bit

lgray committed:
4D vertex validation

felicepantaleo committed:
remove energy from cluster

prebello committed:
fixing typo

smuzaffar committed:
fix clang build error: replace static const with static constexpr

ebrondol committed:
adding LogError

makortel committed:
Fix summary harvesting for built tracks

arunhep committed:
remove LumiPixels from list of AlCaRecos

jalimena committed:
restrict to look at certain PDG ids

rekovic commented:
please test

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/648

Code check has found code style and quality issues which could be resolved by applying a patch in https://cmssdt.cern.ch/SDT/code-checks/PR-20477/648/git-diff.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/PR-20477/648/git-diff.patch | patch -p1`

You can run `scram build code-checks` to apply code checks directly (this will soon be required for PRs to be considered)

davidlange6 commented:
please test

davidlange6 commented:
please test

rekovic committed:
Apply patch from -code-checks.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/650

cmsbuild commented:
A new Pull Request was created by  @rekovic  for master.

It involves the following packages:

DQM/Integration
DQM/L1TMonitor
EventFilter/L1TRawToDigi
L1Trigger/L1TMuon


@vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn can you please review it and eventually sign? Thanks.
@batinkov, @makortel, @Martin-Grunewald, @thomreis, @threus, @kreczko this is something you requested to watch as well.
@davidlange6, @slava77 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

rekovic commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild committed:
Merge pull request #20455 from cms-tsg-storm/93XHLTmenuV3p1

HLT menu "Luna" V3.1 (94X)

cmsbuild committed:
Merge pull request #20427 from bsunanda/Run2-alca94

Run2-alca94 Add a new method to compute energy sum of ECAL towers preceding a HCAl cell…

cmsbuild committed:
Merge pull request #20416 from bsunanda/Run2-hcx151

Run2-hcx151 Correct for geometry access for Hcal

cmsbuild committed:
Merge pull request #20319 from pgunnell/TuneSnippets2017

New tune snippets

kpedro88 committed:
configurable parameter dropUnseedable, disabled by default, enabled for HGCal

kpedro88 committed:
more code checks

cmsbuild commented:
+1
Tested at: c7452ea4e321dc762bad357933a8ba7230f19ccc
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/22905/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/69441fc4eb50c3debac1d7d1d68b2144086d675a
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/22905/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/22905/git-merge-result

cmsbuild commented:
Comparison job queued.

swagata87 committed:
create and clear vector of vertex index per-muon, not per event.

kurtejung committed:
fixed ptMin threshold for PU subtractor

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/22905/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2642440
 - DQMHistoTests: Total failures: 204
 - DQMHistoTests: Total nulls: 81
 - DQMHistoTests: Total successes: 2641966
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

kurtejung committed:
fixing code checks

cmsbuild committed:
Merge pull request #20470 from makortel/fixMTVBuiltTracksHarvesting

Fix summary harvesting for built tracks MTV

cmsbuild committed:
Merge pull request #20469 from ebrondol/qualityTrkDebug

adding LogError for quality computation

cmsbuild committed:
Merge pull request #20454 from kurtejung/HeavyIonJetDQM_93xUpdates

RelVal fix for HI Jet DQM

cmsbuild committed:
Merge pull request #19853 from perrozzi/MINIAODGEN_master

introduce MINIGEN step and output to replace GenOnly workflows

cmsbuild committed:
Merge pull request #20468 from smuzaffar/clang-fx-94x

fix clang build error: replace static const with static constexpr

cmsbuild committed:
Merge pull request #19935 from lgray/topic_vectorize_zt

Vectorized 2D simulated annealing vertexing

gregor-mittag committed:
Fix geometry centering feature and avoid moving reference alignment.

felicepantaleo committed:
recalibrating pfclusters

kpedro88 committed:
remove comment

Commit:
code-checks

gpetruc committed:
Fix PackedCandidate::dz(point) for cases where the track and candidate eta's differ, and point is not the associated PV

gpetruc committed:
PhotonIDValueMapProducer: on miniAOD, get dxy/dz from candidate, not from track (which may not be available)

gpetruc committed:
code-checks

Commit:
Stub builder fix and new SW

Commit:
Fix compilation warnings in CondCore/EcalPlugins

arunhep committed:
update L1T conditions needed for emulator

swagata87 committed:
Andrea's suggestion (1) included

swagata87 committed:
modify constructor

fioriNTU committed:
Use Pixel Phase1 code in HI workflows

fioriNTU committed:
cleaning commented lines

zdemirag committed:
changing to nullptr

Dr15Jones committed:
Fix histogram limits in SiPixelClusterSource

The module SiPixelClusterSource was causing frequent failures in
the nightly integration tests because the histogram tracking
layer 4 did not have properly defined limits in the y dimension.

Dr15Jones committed:
Applied clang-tidy changes

gregor-mittag committed:
Migrate geometry-centering config to phase-1 geometry.

gregor-mittag committed:
Add missing linebreak.

This way one avoids that the warning "hides" other output which was
previously printed on the same line.

gregor-mittag committed:
Apply code-checks.

apana committed:
Fix to prevent possible crashes if the prescale table changes during job

wddgit committed:
Add a comment

apana committed:
Code updated after running scram build code-checks

maojiajing committed:
add client to razor paths

cmsbuild committed:
Merge pull request #20496 from fioriNTU/FixHIPhase1

Fix hi phase1

cmsbuild committed:
Merge pull request #20479 from felicepantaleo/HGCal_fix_energydistance_9_3_0_pre5

HGCal - bugfix - removing energy of out-of-range hits

cmsbuild committed:
Merge pull request #20473 from arunhep/AlCaMatrix_update_removeLumiPixel

remove LumiPixels from list of AlCaRecos from Matrix

cmsbuild committed:
Merge pull request #20458 from kpedro88/SkipEmptyClusters

reject seed candidates not part of superclusters

cmsbuild committed:
Merge pull request #20344 from npostiau/update_SMP_validation_paths

Update smp validation paths

kurtejung committed:
removing fillDescriptors for PU Subtractor

gregor-mittag committed:
Fix static analyzer warnings in 'Alignment/CocoaModel/src/OpticalObject.cc'.

${CMSSW_BASE}/src/Alignment/CocoaModel/src/OpticalObject.cc:291:31: warning: Function call argument is an uninitialized value
    if(ALIUtils::debug >= 5 ) std::cout << " OpticalObject::transformCylindrical2Cartesian " << ii << " " << newcoor[ii] << std::endl;
                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
${CMSSW_BASE}/src/Alignment/CocoaModel/src/OpticalObject.cc:292:5: warning: Function call argument is an uninitialized value
    theCoordinateEntryVector[ii]->setValue( newcoor[ii] );
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.

Commit:
code patches applied

ianna committed:
First step for simulation clang tidy code checks all

ianna committed:
Remove unused variables as reported by compiler when the following flag is set: -Wunused-variable

cmsbuild committed:
Merge pull request #20499 from ghellwig/fix-geometry-centering_94X

[94X] Fix geometry centering tool.

cmsbuild committed:
Merge pull request #20498 from Dr15Jones/fixHistogramLimits

Fix histogram limits in SiPixelClusterSource

cmsbuild committed:
Merge pull request #20309 from cms-met/hcalgeo_93x

propagation of hcal geometry changes for the csc halo algo

cmsbuild committed:
Merge pull request #20444 from wddgit/refactoringForNanoAOD

Move functions filling common provenance to IOPool/Provenance

ggovi committed:
Fix for handling of IBs based on Patch-IBs

rekovic committed:
Introduce alias run2_mc_l1stage1 and adjust configuration used by HLT AddOn tests to use it.

rekovic committed:
Merged refs/pull/20494/head from repository cms-sw with cms-merge-topic

clelange committed:
correctly update boundary box preventing overwriting in case a hit has x value = 0

clelange committed:
apply code checks

ianna committed:
Second step for simulation clang tidy code checks all

cmsbuild committed:
Merge pull request #20461 from prebello/from-CMSSW_9_4_X_2017-09-11-1100

updates in relvals

cmsbuild committed:
Merge pull request #20480 from swagata87/new_scouting94X

create and clear vector of vertex index per-muon, not per event.

cmsbuild committed:
Merge pull request #20449 from fabozzi/from-CMSSW_9_4_X_dqmonly

adding DQM-only wf on AOD input for HLT validation

ianna committed:
Remove unused variable

ianna committed:
Third step for simulation clang tidy code checks all

ianna committed:
Step 4  for simulation clang tidy code checks all

ianna committed:
Step 5 of simulation clang tidy code checks all

jalimena committed:
cleanup

ianna committed:
Step 6 of simulation clang tidy code-checks-all

rekovic committed:
Merged refs/pull/20518/head from repository cms-sw with cms-merge-topic

rekovic committed:
fastsim1 addOnTest to use auto:run2_mc_l1stage1 instead of auto:run2_mc.

ianna committed:
Step 7 of simulation clang tidy code-checks-all

ahinzmann committed:
remove mayConsumes

davidlange6 committed:
Merge pull request #20010 from dimattia/ChargeHistoBinsRed

Memory reduction of the Cluster Charge Histogram used in SiStrip gain calibration

ianna committed:
Step 8 of simulation clang tidy code-checks-all

rekovic commented:
+1

ahinzmann committed:
implement code checks

cmsbuild committed:
Merge pull request #20503 from apana/apana_bugfix_PrescaleCode_94x

Fix to prevent possible crashes if the prescale table changes during job

cmsbuild committed:
Merge pull request #20421 from jalimena/RHStopTracerFix94X

bug fixes for RHStopTracer (94X)

cmsbuild committed:
Merge pull request #20524 from ianna/sim-clang-checks-all-step1-v2

Step 1 amendment: Simulation Clang tidy code-checks-all and Clang Warnings Removed

cmsbuild committed:
Merge pull request #20490 from dildick/from-CMSSW_9_4_X_2017-09-11-2300-PR19986-code-checks

Code checks for PR19986

cmsbuild committed:
Merge pull request #20450 from makortel/seedStopReason

Monitor seed stopping reason and number of trajectory candidates per seed in tracking DQM

gregor-mittag committed:
Get rid of thread-unsafe non-const static local variables in 'MillePedeMonitor'.

forthommel committed:
Removed the unused UFSD materials list (to avoid information duplication) ; started to modify the detectors assembly to include the UFSD plane (now, at position 4...)

forthommel committed:
Reusing the diamond detector assembly to include the UFSD plane ; small correction in the definition of the z size of the strip

forthommel committed:
updated DQM test

forthommel committed:
Updated UFSD geometry, dqm test on recent run

smuzaffar committed:
fixes need for llvm 5.0: use fully qualified clangcms::support

smuzaffar committed:
applied code-checks suggestions

davidlange6 committed:
Merge pull request #20406 from gartung/DQMServices-Core-protobuf-3.4.0

Updates to DQMServices/Core for new protobuf version.

davidlange6 committed:
Revert "Updates to DQMServices/Core for new protobuf version."

forthommel committed:
Integration of the UFSD geometrical elements names in a 9_4_X environment

davidlange6 committed:
Merge pull request #20543 from cms-sw/revert-20406-DQMServices-Core-protobuf-3.4.0

Revert "Updates to DQMServices/Core for new protobuf version."

cmsbuild committed:
Merge pull request #20409 from bsunanda/Run2-hcx150

Run2-hcx150 Correct access to Hcal geometry

forthommel committed:
bug fixing

Conflicts:

	DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

smuzaffar committed:
Revert "Revert "Updates to DQMServices/Core for new protobuf version.""

smuzaffar committed:
apply code-checks

VinInn committed:
increase y limits in profile to avoid saturation

VinInn committed:
code-checks

smuzaffar committed:
Merge pull request #20541 from cms-sw/llvm5-fix-devel-ibs

fixes needed for llvm 5.0: use fully qualified clangcms::support namespace

cmsbuild committed:
Merge pull request #20544 from cms-sw/revert-20543-revert-20406-DQMServices-Core-protobuf-3.4.0

Updates to DQMServices/Core for new protobuf version

cmsbuild committed:
Merge pull request #20527 from ianna/sim-clang-checks-all-step2-v2

Step 2 amendment: Simulation Clang tidy code-checks-all and Clang Warnings Removed

forthommel committed:
Reverted back the DQM modifications

natalia-korneeva committed:
Switching from TString to std::string

natalia-korneeva committed:
Fixing code quality

Commit:
Modify the MIP analyzer in view of new requirements

Commit:
Take care of codechecker comments

gregor-mittag committed:
Fix thread-safety issue (const_cast of 'this' pointer) in 'PedeLabeler'.

gregor-mittag committed:
Use range-based loops with const qualifier in 'PedeLabeler'.

gregor-mittag committed:
Fix thread-safety issue (const_cast of 'this' pointer) in 'RunRangeDependentPedeLabeler'.

This line was actually never executed:
  The if statement always returned true because the reverse map is
  already built in the constructor.

gregor-mittag committed:
Use range-based loops with const qualifier in 'RunRangeDependentPedeLabeler'.

gregor-mittag committed:
Fix thread-safety issue (const_cast of 'this' pointer) in 'MomentumDependentPedeLabeler'.

This line was actually never executed:
  The if statement always returned true because the reverse map is
  already built in the constructor.

cmsbuild committed:
Merge pull request #20516 from ggovi/condcore-utilities-cond2xml-improv-1-94X

Fix for conddb dump executed on IBs

cmsbuild committed:
Merge pull request #20482 from kurtejung/HiJetRecoPtMinFix

fixed ptMin threshold for PU subtractor

cmsbuild committed:
Merge pull request #20445 from rappoccio/checkingptrs_94x

Fixing key comparison in pat::Jet construction for miniAOD

cmsbuild committed:
Merge pull request #20401 from pmillet/sherpa_openmpi_support

bugfix for sherpa compiled with openmpi support

cmsbuild committed:
Merge pull request #20386 from folguera/DQM93X_PixelPhaseI_TrendsByLumiBlockForOffline

PixelPhase1 trends by lumiblock for offline

gregor-mittag committed:
Use range-based loops with const qualifier in 'MomentumDependentPedeLabeler'.

cmsbuild committed:
Merge pull request #20493 from mrodozov/fix-comp-warnings-CondCore/EcalPlugins

Fix compilation warnings in CondCore/EcalPlugins

cmsbuild committed:
Merge pull request #20270 from CTPPS/ctpps-timing_dqm_9_3_X_august2017

Updated DQM after front-end optimization

gregor-mittag committed:
Clean up CVS remnant.

cmsbuild committed:
Merge pull request #19771 from perahrens/CMSSW_9_3_X

Embedding developments: New feature electron embedding and bugfixes

gregor-mittag committed:
Apply code-checks.

folguera committed:
support hitless seeds in GroupedCkfTrajectory

folguera committed:
added code-checks suggestions

folguera committed:
Merge branch 'master' into RecoTracker_MuonHLT_GroupedCkfForHitless

ianna committed:
Merged GEMGeometryBuilderDB-fix from repository jshlee with cms-merge-topic

cmsbuild committed:
Merge pull request #20534 from ianna/sim-clang-checks-all-step8

Step 8: Simulation Clang tidy code-checks-all

Dr15Jones committed:
Properly handle failed file opening in DQMRootSource

Since ROOT 6 is unable to propagate exceptions through cling,
it is necessary to trap file open errors in TFileAdaptor and
then call edm::threadLocalException::getException() to get
the underlying exception to be rethrown.

cmsbuild committed:
Merge pull request #20518 from rekovic/pr-20494-l1stage1-alias

Pr 20494 plus l1stage1 alias

Dr15Jones committed:
Applied clang-tidy changes

ianna committed:
Remove const castness

Commit:
Modify the cfg's

Commit:
Decouple response correction and gain correction parts

Commit:
Adding Run1 JEC and MVA Selectors

Dr15Jones committed:
Handle the HLT case in edmTracerLogToSimpleConfig

-Does not add 'PathStatus' prox modules to configuration
-Ignore 'TriggerResults' in consumes relationship because the type
 does not match in the proxy Int modules

smuzaffar committed:
use ClassDefOverride instead of ClassDef to avoid inconsistent-missing-override warning

Commit:
rename 2017MCTunes folder to MCTunes2017 to avoid import * problems

gregor-mittag committed:
Use thread-safe bool for one-time-warning in 'PedeSteerer'.

gregor-mittag committed:
Apply code-checks.

Commit:
restore usual convention for PSet naming, fix CP1star to CP2 naming

cmsbuild committed:
Merge pull request #20570 from smuzaffar/inconsistent-missing-override-warning

use ClassDefOverride instead of ClassDef to avoid inconsistent-missing-override warning

cmsbuild committed:
Merge pull request #20569 from Dr15Jones/handleHLTCase

Handle the HLT case in edmTracerLogToSimpleConfig

fabferro committed:
returning pixel gain calibration by reference

rekovic commented:
@dmitrijus Please sign/comment.

cmsbuild committed:
Merge pull request #20521 from clelange/FixLayerClusteringBoundingBox_94X

correctly update boundary box preventing overwriting in case a hit ha…

cmsbuild committed:
Merge pull request #20533 from ahinzmann/removeMayConsumes

Remove mayConsumes from PATJetProducer

cmsbuild committed:
Merge pull request #20553 from bsunanda/Run2-alca97

Run2-alca97 Modify the MIP analyzer in view of new requirements

Commit:
Add logic to handle exclusion of CALOL2 in DQM

Commit:
adding prompt like GT

fabferro committed:
returning pixel gain calibration by reference

fabferro committed:
simplifying code

arunhep committed:
Era update and removal of LumiPixels from AlCaLumiPixel, addition of AlCaPCCZeroBias and AlCaPCCRandom

Commit:
fix WF

rekovic committed:
Revert change in GlobalTag introduced while rebasing.

dmitrijus commented:
Please remove changes to:
DQM/Integration/python/config/*

Otherwise it is fine and can go in.

cmsbuild commented:
The code-checks are being triggered in jenkins.

dmitrijus commented:
Ah! DQM/Integration/python/config/FrontierCondition_GT_cfi.py still remains.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/819

cmsbuild commented:
Pull request #20477 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn can you please check and sign again.

rekovic committed:
Revert change in the second GlobalTag introduced while rebasing.

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic committed:
Revert change in empty space.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/820

cmsbuild commented:
Pull request #20477 was updated. @vazzolini, @kmaeshima, @dmitrijus, @cmsbuild, @rekovic, @vanbesien, @mulhearn can you please check and sign again.

rekovic commented:
please test

rekovic commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

rekovic commented:
@dmitrijus done.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/822

dmitrijus commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request will now be reviewed by the release team before it's merged. @davidlange6, @slava77, @smuzaffar (and backports should be raised in the release meeting by the corresponding L2)

cmsbuild committed:
Merge pull request #20572 from perrozzi/MCTunes2017_rename_master

rename 2017MCTunes folder to MCTunes2017 to avoid import * problems

cmsbuild committed:
Merge pull request #20562 from ianna/gem-builders-const-cast-fix

GEM Geometry Builders Remove const_cast

cmsbuild committed:
Merge pull request #20560 from Dr15Jones/handleFileOpenErrorInDQMRootSource

Properly handle failed file opening in DQMRootSource

cmsbuild committed:
Merge pull request #20559 from ghellwig/fix-thread-safety-warnings_AliMP

[94X] Fix thread safety warnings in 'Alignment/MillePedeAlignmentAlgorithm'.

cmsbuild committed:
Merge pull request #20549 from natalia-korneeva/hlt-btag-std-string

Switching from TString to std::string

cmsbuild committed:
Merge pull request #20545 from VinInn/TkDQM94_0

increase y limits in profile to avoid saturation for TRK DQM

cmsbuild committed:
Merge pull request #20512 from ghellwig/fix-static-anaylzer-warnings-OpticalObject

Fix static analyzer warnings in 'Alignment/CocoaModel/src/OpticalObject.cc'

cmsbuild committed:
Merge pull request #20492 from sviret/Stub_builder_fix_9_4

Stub builder fix and new SW

thomreis committed:
Move import of cff before the use of the sequence.

cmsbuild commented:
+1
Tested at: ace3e37aa68f6a0f5f6964498ba3d5a3177e8759
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/23082/summary.html

The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/5e53c2b9539aa99597276fc2af455b7acffca3cf
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/23082/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/23082/git-merge-result

cmsbuild commented:
Comparison job queued.

arunhep committed:
json update for the SiPixelAli

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-20477/23082/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 26
 - DQMHistoTests: Total histograms compared: 2649068
 - DQMHistoTests: Total failures: 228
 - DQMHistoTests: Total nulls: 81
 - DQMHistoTests: Total successes: 2648570
 - DQMHistoTests: Total skipped: 189
 - DQMHistoTests: Total Missing objects: 0
 - Checked 107 log files, 14 edm output root files, 26 DQM output files

mackoo13 committed:
UFSD Geometry fix (with Laurent and Nicola)

cmsbuild committed:
Merge pull request #20262 from mschrode/fix_GBL_missing_initialisations_93X

FIX in GBL reference trajectories: add missing initialisations

cmsbuild committed:
Merge pull request #20508 from maojiajing/RazorDQM_upgrade_94x

add client to razor paths (PR for master/94X)

forthommel committed:
Cleanup of UFSD and diamond DDL comments

cmsbuild committed:
Merge pull request #20579 from ashtipliyski/calol2-online-dqm-94x

Add logic to handle exclusion of CALOL2 in DQM

Commit:
Cleanup

ianna committed:
Replace boost lexical cast with std

cmsbuild committed:
Merge pull request #20580 from arunhep/EraUpdate_AlCaLumiPixels_94X

Era update and removal of LumiPixels from AlCaLumiPixel with addition of AlCaPCCZeroBias and AlCaPCCRandom

Commit:
Make protection in the analyzer

Commit:
Take care of comments

cmsbuild committed:
Merge pull request #20601 from ianna/gem-builders-lexical-cast-fix

Replace boost::lexical_cast<std::string> with std::to_string

cmsbuild committed:
Merge pull request #20429 from makortel/cleanupTrackMVA

Refactor track MVA classifier code

cmsbuild committed:
Merge pull request #20313 from thomreis/l1t_offline_dqm_express_stream_harvesting

Add L1 offline DQM harvesting step

davidlange6 commented on EventFilter/L1TRawToDigi/interface/BxBlock.h:
```diff
@@ -0,0 +1,70 @@
+#ifndef EventFilter_L1TRawToDigi_BxBlock_h
```
hi @rekovic - this class is a data format - please move it to a DataFormats package

davidlange6 replied:
(as are other classes in this package, but lets at least not add new ones)

cmsbuild committed:
Merge pull request #20491 from gpetruc/fix-dz-and-PhotonIDValueMapProducer

Fix PackedCandidate::dz and PhotonIDValueMapProducer

cmsbuild committed:
Merge pull request #20548 from forthommel/ctpps-ufsd_geometry-9_4_X

CTPPS: Geometry for UFSDetectors

cmsbuild committed:
Merge pull request #20605 from bsunanda/RUn2-alca98

RUn2-alca98 Make protection in the analyzer

cmsbuild committed:
Merge pull request #20593 from arunhep/DropboxUpdate_SiPixelAli_backToProd

Dropbox json update for the SiPixelAli for allowing to write in production tag

cmsbuild committed:
Merge pull request #20578 from CTPPS/ctpps_pixel_cluster_patch

CTPPS pixel gain calibration patch

cmsbuild committed:
Merge pull request #20568 from lpernie/RunJec_MVAselec

GT Updates: Run1 JEC and MVA Selectors as Tracking MVA Selectors update

cmsbuild committed:
Merge pull request #20556 from folguera/RecoTracker_MuonHLT_GroupedCkfForHitless

Support hitless seeds in GroupedCkfTrajectory

rekovic committed:
Merged refs/pull/20477/head from repository cms-sw with cms-merge-topic

rekovic committed:
Move class BxBlock to DataFormats/L1Trigger.

cmsbuild commented:
The code-checks are being triggered in jenkins.

rekovic commented:
will rebase

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/PR-20477/929

cmsbuild commented:
Pull request #20477 was updated. @ghellwig, @lveldere, @ianna, @kpedro88, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @govoni, @perrotta, @civanch, @monttj, @silviodonato, @cmsbuild, @fwyzard, @GurpreetSinghChahal, @davidlange6, @smuzaffar, @Dr15Jones, @kkousour, @ssekmen, @mdhildreth, @cerminar, @slava77, @ggovi, @mulhearn, @efeyazgan, @vazzolini, @kmaeshima, @arunhep, @perrozzi, @thuer, @dmitrijus, @franzoni, @lpernie can you please check and sign again.
